src/wrap.js

```diff
@@ -63,12 +63,15 @@ jQuery.fn.extend({
 		});
 	},
 
-	unwrap: function() {
-		return this.parent().each(function() {
-			if ( !jQuery.nodeName( this, "body" ) ) {
-				jQuery( this ).replaceWith( this.childNodes );
-			}
-		}).end();
+	unwrap: function( selector ) {
+		return this
+			.parent()
+				.filter( selector || ":not(body)" )
```

Comment:
Yeah, I agonized on that one and figured a selector that included body would be pretty rare. Your solution is safer, I'll play with this.